### PR TITLE
NSStackView

### PR DESCRIPTION
**Pull request by gcasa**
Changes to implement NSStackView

### CODE REVIEW

File: Source/NSStackView.m [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+  RELEASE(_gravityMap);
+  RELEASE(_visiblePriorityMap);
+  RELEASE(_clippingMap);
+  _delegate = nil;
+  [super dealloc];
+}
+
++ (instancetype) stackViewWithViews: (NSArray *)views
+{
+  return [[self alloc] initWithViews: views];
```
Comment (fredkiefer) [inline Source/NSStackView.m:154]: AUTORELEASE is missing here.

Comment (gcasa) [reply Source/NSStackView.m:154]: You're correct.  Will fix this.   Welcome back.

**Comment by gcasa** [comment]
Here is the enumerated nib to support the keys used in the code.
[MainMenu-enumerated.nib.txt](https://github.com/gnustep/libs-gui/files/6498062/MainMenu-enumerated.nib.txt)

File: Source/GSXib5KeyedUnarchiver.m [hunk 2]
```diff
@@ -3044,6 +3053,62 @@ - (id) decodeYPlacementForElement: (GSXibElement *)element
...
+    {
+      d = NSStackViewDistributionFillEqually;
+    }
+  else if ([obj isEqualToString: @"fillProportionally"])
+    {
+      d = NSStackViewDistributionFillProportionally;
+    }
+  else if ([obj isEqualToString: @"gravityAreas"])
+    {
+      d = NSStackViewDistributionEqualCentering;
```
Comment (fredkiefer) [inline Source/GSXib5KeyedUnarchiver.m:3101]: Shouldn't this be NSStackViewDistributionGravityAreas?

Comment (gcasa) [reply Source/GSXib5KeyedUnarchiver.m:3101]: No, in the xib these are the values for that property.   The test for this, by the way, is at https://github.com/gcasa/NSStackView_test.

**Comment by gcasa** [comment]
Here is the XIB file just for reference.
[MainMenu.xib.txt](https://github.com/gnustep/libs-gui/files/6538854/MainMenu.xib.txt)

**Comment by gcasa** [comment]
@fredkiefer Please review.  NOTE: I am not implementing hugging priorities and customSpacing in this iteration.

**Comment by gcasa** [comment]
Please see attached images...

NSStackView_linux
NSStackView_macOS

**Pull request merged**